internals/cli/ui/io_windows.go

```diff
@@ -1,24 +1,32 @@
 package ui
 
 import (
+	"io"
 	"os"
 
 	"github.com/fatih/color"
 	colorable "github.com/mattn/go-colorable"
 	isatty "github.com/mattn/go-isatty"
 )
 
-// NewUserIO creates a new UserIO middleware from os.Stdin and os.Stdout and adds tty if it is available.
-func NewUserIO() UserIO {
-	// Ensure colors are printed correctly on Windows.
-	if !color.NoColor {
-		return UserIO{
-			input:  os.Stdin,
-			output: colorable.NewColorable(os.Stdout),
-		}
+// windowsIO is the Windows-specific implementation of the IO interface.
+type windowsIO struct {
+	standardIO
+}
+
+// NewUserIO creates a new windowsIO.
+func NewUserIO() IO {
+	return windowsIO{
+		standardIO: newStdUserIO(),
 	}
+}
 
-	return NewStdUserIO()
+// Stdout returns the standardIO's Output.
+func (o windowsIO) Stdout() io.Writer {
+	if !color.NoColor {
+		return colorable.NewColorable(os.Stdout)
+	}
```

Comment:
One difference is that this is now evaluated every time `Stdout` is called. Is this a lightweight operation?

Reply:
Good point. Doing this at init now.